backends/fs/psp2/psp2-fs.cpp

```diff
@@ -0,0 +1,252 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#if defined(PSP2)
+
+#define FORBIDDEN_SYMBOL_EXCEPTION_time_h
+
+#define FORBIDDEN_SYMBOL_EXCEPTION_unistd_h
+
+#define FORBIDDEN_SYMBOL_EXCEPTION_mkdir
+
+#include "backends/fs/psp2/psp2-fs.h"
+#include "backends/fs/psp2/psp2-stream.h"
+#include "common/bufferedstream.h"
+#include "common/debug.h"
+#include "engines/engine.h"
+
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include <psp2/io/dirent.h>
+#include <psp2/io/stat.h>
+
+
+PSP2FilesystemNode::PSP2FilesystemNode(const Common::String &path, const Config &config, bool verify) :
+        _isPseudoRoot(false),
+	    _config(config) {
+
+    _path = path;
+    _displayName = lastPathComponent(_path, '/');
+
+    if (isDrive(path)) {
+        _isDirectory = true;
+        _isValid = true;
+
+        if (!_path.hasSuffix("/")) {
+			_path += "/";
+		}
+
+        debug(8, "path [%s]", _path.c_str());
+
+        if (verify) {
+            struct stat st;
+            if (PowerMan.beginCriticalSection() == PowerManager::Blocked)
+                debug(8, "Suspended");
+            _isValid = (0 == stat(_path.c_str(), &st));
```

Comment:
Since this file is already using native APIs instead of dirent functions, it might be worth replacing `stat` and `access` calls as well.